Mathlib/Tactic/CancelDenoms/Core.lean

```diff
@@ -328,13 +328,13 @@ open Elab Tactic
 
 def cancelDenominatorsAt (fvar : FVarId) : TacticM Unit := do
   let t ← instantiateMVars (← fvar.getDecl).type
-  let (new, eqPrf) ← CancelDenoms.cancelDenominatorsInType t
+  let (new, eqPrf) ← CancelDenoms.cancelDenominatorsInType t.consumeMData --  unneeded?
```

Comment:
I think the flaw here is actually in `findCompLemma`, which should be doing a `whnfR`

Reply:
`findCompLemma` is not monadic, though: I simply made it recursively remove `mdata`.

Reply:
Moved to #12083.